mlflow/pyfunc/__init__.py

```diff
@@ -2327,205 +2327,210 @@ def predict(model_input: List[str]) -> List[str]:
             .. Note:: Experimental: This parameter may change or be removed in a future
                                     release without warning.
     """
-    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
-    _validate_pyfunc_model_config(model_config)
-    _validate_and_prepare_target_save_path(path)
-
-    model_code_path = None
-    if python_model:
-        if isinstance(model_config, Path):
-            model_config = os.fspath(model_config)
-
-        if isinstance(model_config, str):
-            model_config = _validate_and_get_model_config_from_file(model_config)
-
-        if isinstance(python_model, Path):
-            python_model = os.fspath(python_model)
+    with tempfile.TemporaryDirectory() as temp_dir:
+        _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+        _validate_pyfunc_model_config(model_config)
+        _validate_and_prepare_target_save_path(path)
+
+        model_code_path = None
+        if python_model:
+            if isinstance(model_config, Path):
+                model_config = os.fspath(model_config)
+
+            if isinstance(model_config, str):
+                model_config = _validate_and_get_model_config_from_file(model_config)
+
+            if isinstance(python_model, Path):
+                python_model = os.fspath(python_model)
+
+            if isinstance(python_model, str):
+                model_code_path = _validate_and_get_model_code_path(python_model, temp_dir)
```

Comment:
same as in langchain - only change is passing in temp_dir here (rest is indentation)

Reply:
Are there any lines that can be moved outside of `with tempfile.TemporaryDirectory`? For example, `_validate_env_arguments` doesn't have to be in it.

Reply:
Yes - I'll lower it to where it is first needed.